### ISSUE TITLE
Queries For Potential False Positives on Accessibility Insights For Android

### ISSUE THREAD

**Issue by daphneyyt** [microsoft/accessibility-insights-web#3862]
Hello, I would like to seek clarification on potential false positives from Accessibility Insights For Android's findings/ reports.

For context: We have done mobile apps scanning with Accessibility Insights and have sent our reports to the app owners for their review and follow-up. 

**1. Scenario 1: WCAG 1.3.1, 3.3.2, Accessibility Insights had flagged the following as an ActiveViewName issue (please see images and report attached)**

Accessibility Insight Scan Part 1: 
<img width="789" alt="WCAG 131 332 - 1" src="https://user-images.githubusercontent.com/75086091/106594161-af8cd000-658c-11eb-8541-099c580d4d54.png">

Accessibility Insight Scan Part 2: 
<img width="787" alt="WCAG 131 332 - 2" src="https://user-images.githubusercontent.com/75086091/106594165-b1569380-658c-11eb-9dbc-2acf723c8181.png">

Accessibility Insight Scan Report: Please see Accessibility Insights for Android - Report 131 332.PDF
[Accessibility Insights for Android - Report 131 332.pdf](https://github.com/microsoft/accessibility-insights-for-android-service/files/5910577/Accessibility.Insights.for.Android.-.Report.131.332.pdf)

**App Owner's response**
The app owner informed us that "the page on the app use WebView (i.e. the mobile app wraps a HTML webpage in the backend). The fields and links highlighted in the report already contain name and label in all the HTML tags. These are a false positives." 
![SP WCAG 131 332](https://user-images.githubusercontent.com/75086091/106590970-a994f000-6588-11eb-8a0f-eb3de36d693f.png)

**Our Checks**
As this set of codes are codes from the website that is appearing in the mobile app, we used Inspect Element on Google to check the codes as well. We note that names and labels are indeed provided within their codes.
<img width="792" alt="WCAG 131 332 label" src="https://user-images.githubusercontent.com/75086091/106592286-4a37df80-658a-11eb-9b4c-54abbdb00cef.png">

**We would like to know:**
1) apart from label tags, could there be anything else that Accessibility Insights could be flagging about?
2) What aspect of this label tag should be adjusted on the app owner's end to pass this WCAG 1.3.1/ 3.3.2 criteria?
3) Are these indeed false positives flagged by Accessibility Insights?

**2. Scenario 2: WCAG 1.4.3, Accessibility Insights had flagged the following as a colorcontrast issue**
Accessibility Insight Scan: 
<img width="789" alt="WCAG 143 - 1" src="https://user-images.githubusercontent.com/75086091/106594042-866c3f80-658c-11eb-89a8-d58763052795.png">

**App owner's response**
App owner said that the "Highlighted items are images. The text that failed the contract ratio as mentioned in the report is an “alt” text, so this should not be a finding because we have no control over the colour of the “alt” text."

**We would like to know:**
1) Is this considered a false-positive raised by Accessibility Insights?
2) What aspect of this alt text should be adjusted on the app owner's end to pass this WCAG 1.4.3 criteria?

Please feel free to reach out for more details is needed. However, I am unable to share the app name/ details with the team as personal login details are required to access this app, and the information is confidential.

Thank you!

--

**Comment by pownkel** [comment]
Hi @daphneyyt, thank you for using Accessibility Insights for Android! Since this is not a bug in the service, I've moved it into the repo for our Android app and web extension.

1. **Scenario 1:**  Since this uses a WebView and looks like a web application, we recommend testing with [Accessibility Insights for Web](https://accessibilityinsights.io/docs/en/web/overview/) instead, using the browser's "Toggle device toolbar" option to toggle the web app to an appropriate Android device resolution if that is possible. We also suggest testing using AT(screen reader), as the web attributes might work differently in an Android application.

2. **Scenario 2:** This is not considered a false positive, because it is shown under "Needs Review." The "Needs Review" panel identifies issues that _might_ be accessibility bugs, and need to be reviewed by a human. In this case, the "how to check" information states, "if the text is invisible, it passes." Since this is invisible alt text, it is not a problem for accessibility.

Let us know if this helps!

**Comment by daphneyyt** [comment]
Hi @pownkel thank you for your response! 

 Noted on the issue for WCAG 1.3.1, 3.3.2 - we will use Accessibility Insights For Web for more accurate findings. 

I have a follow-up question:
1) **WCAG 1.4.3** - Noted that there is no issues for the alt text as it is invisible, and was flagged by the Accessibility Insights For Android as a potential finding. However, since this page also uses WebView, can I check whether this finding flagged by Accessibility Insights For Android is still trustworthy?

I will add on to this chat if I have further queries. 

Thank you!

**Comment by pownkel** [comment]
Hi @daphneyyt, while it would not hurt to review the items marked "needs review" by Accessibility Insights for Android, you will generally get more relevant results for a WebView using Accessibility Insights for Web.

**Comment by daphneyyt** [comment]
Hi @pownkel noted. Thank you very much for the clarification! Appreciate it :) 

I do not have further questions at this point.

**Comment by pownkel** [comment]
@daphneyyt Ok, I will mark this issue as resolved. Don't hesitate to ask if you need any additional clarifications from us. Otherwise, if the original issue has been addressed, please close this tracking issue.

**Comment by daphneyyt** [comment]
Noted. Thank you for your help @pownkel ! Appreciate it!
